### PR TITLE
Added support for MOES Zigbee Thermostat (ZHT-002-GC-WH-MS aka beca_zht002)

### PR DESCRIPTION
**Pull request by pandatautau**
First things first! Thank you so much for your amazing work!

Added configuration file for MOES Zigbee thermostat ZHT-002-GC-WH-MS

Note: 
- On the secondary entities, I tried to add a `select` entity to create a temperature sensor selector.
I was not able to, so as a workaround, I instead added a `climate` entity. 
It is definitely not the correct way of showcasing that option (UI-wise), but the only one I was able to implement.

Screenshot of UI

Screenshot 2023-11-13 at 11 22 10

Screenshot 2023-11-13 at 11 23 12

Base data used for the implementation:
```
{
    "result": {
        "model": {
            "modelId": "000001vrud",
            "services": [
                {
                    "actions": [],
                    "code": "",
                    "description": "",
                    "events": [],
                    "name": "默认服务",
                    "properties": [
                        {
                            "abilityId": 1,
                            "accessMode": "rw",
                            "code": "switch",
                            "description": "",
                            "extensions": {
                                "iconName": "icon-dp_power3"
                            },
                            "name": "开关",
                            "typeSpec": {
                                "type": "bool",
                                "typeDefaultValue": false
                            }
                        },
                        {
                            "abilityId": 2,
                            "accessMode": "rw",
                            "code": "mode",
                            "description": "",
                            "extensions": {
                                "attribute": "128"
                            },
                            "name": "工作模式",
                            "typeSpec": {
                                "range": [
                                    "Hand",
                                    "Program"
                                ],
                                "type": "enum",
                                "typeDefaultValue": "Hand"
                            }
                        },
                        {
                            "abilityId": 3,
                            "accessMode": "ro",
                            "code": "work_state",
                            "description": "",
                            "extensions": {
                                "attribute": "128"
                            },
                            "name": "工作状态",
                            "typeSpec": {
                                "range": [
                                    "hot"
                                ],
                                "type": "enum",
                                "typeDefaultValue": "hot"
                            }
                        },
                        {
                            "abilityId": 16,
                            "accessMode": "rw",
                            "code": "temp_set",
                            "description": "",
                            "extensions": {
                                "iconName": "icon-dp_temp"
                            },
                            "name": "目标温度",
                            "typeSpec": {
                                "max": 45,
                                "min": 0,
                                "scale": 0,
                                "step": 1,
                                "type": "value",
                                "typeDefaultValue": 0,
                                "unit": "℃"
                            }
                        },
                        {
                            "abilityId": 18,
                            "accessMode": "rw",
                            "code": "upper_temp_f",
                            "description": "",
                            "extensions": {
                                "iconName": "icon-dp_f"
                            },
                            "name": "高温保护",
                            "typeSpec": {
                                "max": 70,
                                "min": 45,
                                "scale": 0,
                                "step": 1,
                                "type": "value",
                                "typeDefaultValue": 45,
                                "unit": "℃"
                            }
                        },
                        {
                            "abilityId": 19,
                            "accessMode": "rw",
                            "code": "upper_temp",
                            "description": "",
                            "extensions": {
                                "iconName": "icon-dp_c"
                            },
                            "name": "高温保护",
                            "typeSpec": {
                                "max": 70,
                                "min": 45,
                                "scale": 0,
                                "step": 1,
                                "type": "value",
                                "typeDefaultValue": 45,
                                "unit": "℃"
                            }
                        },
                        {
                            "abilityId": 20,
                            "accessMode": "rw",
                            "code": "lower_temp_f",
                            "description": "",
                            "extensions": {
                                "iconName": "icon-dp_f"
                            },
                            "name": "盲区温度",
                            "typeSpec": {
                                "max": 5,
                                "min": 0,
                                "scale": 0,
                                "step": 1,
                                "type": "value",
                                "typeDefaultValue": 0,
                                "unit": "℃"
                            }
                        },
                        {
                            "abilityId": 24,
                            "accessMode": "ro",
                            "code": "temp_current",
                            "description": "",
                            "extensions": {
                                "iconName": "icon-wendu"
                            },
                            "name": "当前温度",
                            "typeSpec": {
                                "max": 700,
                                "min": 0,
                                "scale": 1,
                                "step": 1,
                                "type": "value",
                                "typeDefaultValue": 0,
                                "unit": "℃"
                            }
                        },
                        {
                            "abilityId": 26,
                            "accessMode": "rw",
                            "code": "lower_temp",
                            "description": "",
                            "extensions": {
                                "iconName": "icon-dp_c"
                            },
                            "name": "盲区温度",
                            "typeSpec": {
                                "max": 5,
                                "min": 1,
                                "scale": 0,
                                "step": 1,
                                "type": "value",
                                "typeDefaultValue": 1,
                                "unit": "℃"
                            }
                        },
                        {
                            "abilityId": 27,
                            "accessMode": "rw",
                            "code": "temp_correction",
                            "description": "",
                            "extensions": {
                                "iconName": "icon-wendu1"
                            },
                            "name": "温度校准",
                            "typeSpec": {
                                "max": 9,
                                "min": -9,
                                "scale": 0,
                                "step": 1,
                                "type": "value",
                                "typeDefaultValue": -9,
                                "unit": ""
                            }
                        },
                        {
                            "abilityId": 36,
                            "accessMode": "ro",
                            "code": "valve_state",
                            "description": "",
                            "extensions": {
                                "iconName": "icon-tcl_function_vs2"
                            },
                            "name": "阀门状态",
                            "typeSpec": {
                                "range": [
                                    "open",
                                    "close"
                                ],
                                "type": "enum",
                                "typeDefaultValue": "open"
                            }
                        },
                        {
                            "abilityId": 40,
                            "accessMode": "rw",
                            "code": "child_lock",
                            "description": "",
                            "extensions": {
                                "iconName": "icon-dp_lock"
                            },
                            "name": "童锁",
                            "typeSpec": {
                                "type": "bool",
                                "typeDefaultValue": false
                            }
                        },
                        {
                            "abilityId": 43,
                            "accessMode": "rw",
                            "code": "sensor_choose",
                            "description": "",
                            "extensions": {
                                "iconName": "icon-gongneng"
                            },
                            "name": "传感器选择",
                            "typeSpec": {
                                "range": [
                                    "in",
                                    "al",
                                    "ou"
                                ],
                                "type": "enum",
                                "typeDefaultValue": "in"
                            }
                        },
                        {
                            "abilityId": 101,
                            "accessMode": "rw",
                            "code": "week_program11",
                            "description": "",
                            "name": "周程序",
                            "typeSpec": {
                                "maxlen": 128,
                                "type": "raw"
                            }
                        }
                    ]
                }
            ]
        }
    }
}
```

MOES_zht002

### CODE REVIEW

File: custom_components/tuya_local/devices/beca_zht002_thermostat_c.yaml [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+secondary_entities:
+  - entity: lock
+    name: Child lock
+    icon: "mdi:hand-back-right-off"
+    category: config
+    dps:
+      - id: 40
+        type: boolean
+        name: lock
+  - entity: climate
```
Comment (make-all) [inline custom_components/tuya_local/devices/beca_zht002_thermostat_c.yaml:50]: Change the entity to select, and the name of the preset_mode dps below to option.

File: custom_components/tuya_local/devices/beca_zht002_thermostat_c.yaml [hunk 2]
```diff
@@ -0,0 +1,93 @@
+name: Zigbee Thermostat
+products:
+  - id: bf63833d92f517329fqkhz
```
Comment (make-all) [inline custom_components/tuya_local/devices/beca_zht002_thermostat_c.yaml:3]: Is this the product id? Zigbee devices usually have a shorter product id.

File: custom_components/tuya_local/devices/beca_zht002_thermostat_c.yaml [hunk 3]
```diff
@@ -0,0 +1,93 @@
+name: Zigbee Thermostat
```
Comment (make-all) [inline custom_components/tuya_local/devices/beca_zht002_thermostat_c.yaml:1]: Leave out Zigbee from the top level name, as they may make WiFi and Bluetooth versions that match the same config.

File: custom_components/tuya_local/devices/beca_zht002_thermostat_c.yaml (hunk 2 above)
Comment (pandatautau) [reply custom_components/tuya_local/devices/beca_zht002_thermostat_c.yaml]: Got it, since the `product_id` was so short I was not sure

**Comment by pandatautau** [comment]
@make-all Sorry for the delay in the feedback.

**Pull request merged**